### PR TITLE
Fix/delete object api

### PR DESCRIPTION
**Pull request by lazyvar**
Updated deleteObject API to match README and other functions

### CODE REVIEW

**Comment by lazyvar** [comment]
The keychain group fix got tangled in this too

File: TinyKeychain/Keychain.swift [hunk 1]
```diff
@@ -190,7 +190,7 @@ public struct Keychain {
      - returns: Either success, or a `Keychain.DeletionError`.
      */
     @discardableResult
-    public func deleteObject<O>(withKey key: Key<O>) -> KeychainResult<Void, DeletionError> {
+    public func deleteObject<O>(forKey key: Key<O>) -> KeychainResult<Void, DeletionError> {
```
Comment (swain) [inline TinyKeychain/Keychain.swift:193]: We could bike-shed on this. However, I think merging is the right idea, as more activity on this GitHub repo should/could lead to more people using the framework. Which, in the long term, means more power and fame for you and me.

**Review by swain — APPROVED** [review]
This is an API-breaking change, so we'll need to increment from version 0.0.12 -> 0.1. But, may make sense to just go to 1.0? What do you think @lazyvar? I have held off because we don't have unit tests as of yet.

File: TinyKeychain/Keychain.swift (hunk 1 above)
Comment (lazyvar) [reply TinyKeychain/Keychain.swift:193]: Nothing better than for glory! Was going for consistency here. Either they are all ```with``` or all ```for```.

**Pull request merged**